### PR TITLE
Verify the driver in the BSP of the ART-PI main branch of rtthread.

### PR DESCRIPTION
**Pull request by Rbb666**
## 拉取/合并请求描述：(PR description)

[
Verify the SPI, SPI_FLASH and SPI_FS drivers in the BSP of the art-pi main branch of rtthread. The corresponding configuration is added in kconfig. In sconscript, add the script to generate the. C file in the port folder, and add the driver options of peripherals in cubemx.
]

以下的内容不应该在提交PR时的message修改，修改下述message，PR会被直接关闭。请在提交PR后，浏览器查看PR并对以下检查项逐项check，没问题后逐条在页面上打钩。
The following content must not be changed in the submitted PR message. Otherwise, the PR will be closed immediately. After submitted PR, please use a web browser to visit PR, and check items one by one, and ticked them if no problem.

### 当前拉取/合并请求的状态 Intent for your PR

必须选择一项 Choose one (Mandatory):

- [ ] 本拉取/合并请求是一个草稿版本 This PR is for a code-review and is intended to get feedback
- [x] 本拉取/合并请求是一个成熟版本 This PR is mature, and ready to be integrated into the repo

### 代码质量 Code Quality：

我在这个拉取/合并请求中已经考虑了 As part of this pull request, I've considered the following:

- [x] 已经仔细查看过代码改动的对比 Already check the difference between PR and old code
- [x] 代码风格正确，包括缩进空格，命名及其他风格 Style guide is adhered to, including spacing, naming and other styles
- [x] 没有垃圾代码，代码尽量精简，不包含`#if 0`代码，不包含已经被注释了的代码 All redundant code is removed and cleaned up
- [x] 所有变更均有原因及合理的，并且不会影响到其他软件组件代码或BSP All modifications are justified and not affect other components or BSP
- [x] 对难懂代码均提供对应的注释 I've commented appropriately where code is tricky
- [x] 本拉取/合并请求代码是高质量的 Code in this PR is of high quality
- [x] 本拉取/合并使用[formatting](https://github.com/mysterywolf/formatting)等源码格式化工具确保格式符合[RT-Thread代码规范](../documentation/coding_style_cn.md) This PR complies with [RT-Thread code specification](../documentation/coding_style_en.txt)

### CODE REVIEW

**Comment by Guozhanxin** [comment]
你本地scons编译下试试，看是不是编不过

File: bsp/stm32/stm32h750-artpi-h750/board/port/drv_sdio.c [hunk 1]
```diff
@@ -0,0 +1,546 @@
...
+        sdio->sdio_des.hw_sdio = (struct stm32_sdio *)SDIO2_BASE_ADDRESS;
+        rt_event_init(&sdio->event, "sdio2", RT_IPC_FLAG_FIFO);
+    }
+
+    /* set host default attributes */
+    host->ops = &ops;
+    host->freq_min = 400 * 1000;
+    host->freq_max = SDIO_MAX_FREQ;
+    host->valid_ocr = VDD_32_33 | VDD_33_34;/* The voltage range supported is 3.2v-3.4v */
+//#ifndef SDIO_USING_1_BIT
```
Comment (Guozhanxin) [inline bsp/stm32/stm32h750-artpi-h750/board/port/drv_sdio.c:448]: 注释的代码删掉吧

File: bsp/stm32/stm32h750-artpi-h750/board/port/drv_sdio.c [hunk 2]
```diff
@@ -0,0 +1,546 @@
...
+    {
+        LOG_E("host create fail");
+        return RT_NULL;
+    }
+
+    #endif
+
+    #ifdef BSP_USING_SDIO2
+    //sdmmc2 wifi
+    struct stm32_sdio_des sdio_des2;
```
Comment (Guozhanxin) [inline bsp/stm32/stm32h750-artpi-h750/board/port/drv_sdio.c:514]: 这部分和通用驱动差别很大是吗

Comment (Rbb666) [reply bsp/stm32/stm32h750-artpi-h750/board/port/drv_sdio.c:514]: 是的，差别很大，ART-PI有两个SDIO，驱动有所增加和更改。

**Review by Guozhanxin — CHANGES_REQUESTED** [review]
FTP-lib和otalib就不要加进来了，

File: bsp/stm32/stm32h750-artpi-h750/board/port/wlan_wiced_lib/SConscript [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+LIBS    = []
+LIBPATH = []
+src = []
+
+LIBPATH = [cwd]
+if rtconfig.CROSS_TOOL == 'gcc':
+    LIBS += ['wifi_6212_armcm7_2.1.2_gcc']
+elif rtconfig.CROSS_TOOL == 'keil':
+    LIBS += ['libwifi_6212_armcm7_2.1.2_armcc']
```
Comment (Guozhanxin) [inline bsp/stm32/stm32h750-artpi-h750/board/port/wlan_wiced_lib/SConscript:14]: 库文件没有提交上来

**Review by Guozhanxin — APPROVED** [review]

**Pull request merged**